packages/roosterjs-editor-api/lib/format/setAlignment.ts

```diff
@@ -1,36 +1,146 @@
+
+/**
+ * Check if the element at the cursor is a table or text element
+ * @param element
+ * @returns
+ */
+function isATableOrText(element: HTMLElement, editor: IEditor) {
+    if (!element) {
+        return;
+    }
+    const selection = editor.getSelectionRangeEx();
+    const selectionType = selection.type;
+    if (selection && selectionType === SelectionRangeTypes.Normal) {
+        return TEXT;
+    } else if (
+        editor.isFeatureEnabled(ExperimentalFeatures.TableAlignment) &&
+        selection &&
+        selectionType === SelectionRangeTypes.TableSelection &&
+        isWholeTableSelected(selection)
+    ) {
+        return TABLE;
+    }
+}
+
+/**
+ * Align the element left, center and right
+ * @param editor
+ * @param element the element being aligned
+ * @param elementType type text or table
+ * @param alignment the alignment type
+ * @param addUndoSnapshot check if this function is being called by addUndoSnapshot
+ */
+function alignElement(
+    editor: IEditor,
+    element: HTMLElement,
+    elementType: string,
+    alignment: Alignment,
+    addUndoSnapshot?: boolean
+) {
+    if (elementType === TABLE) {
+        alignTable(editor, element, alignment);
+    } else {
+        alignText(editor, element, alignment, addUndoSnapshot);
+    }
+}
 
+/**
+ * Align text using the margins
+ * @param editor
+ * @param element
+ * @param alignment
+ * @param addUndoSnapshot
+ * @returns
+ */
+function alignTable(editor: IEditor, element: HTMLElement, alignment: Alignment) {
+    if (alignment == Alignment.Center) {
+        element.style.marginLeft = 'auto';
+        element.style.marginRight = 'auto';
+    } else if (alignment == Alignment.Right) {
+        element.style.marginLeft = 'auto';
+        element.style.marginRight = '';
+    } else {
+        element.style.marginLeft = '';
+        element.style.marginRight = 'auto';
+    }
+}
+
+/**
+ * Align text using the text-align
+ * @param editor
+ * @param element
+ * @param alignment
+ * @param addUndoSnapshot
+ * @returns
+ */
+function alignText(
+    editor: IEditor,
+    element: HTMLElement,
+    alignment: Alignment,
+    addUndoSnapshot?: boolean
+) {
+    let align = 'left';
+    let command = DocumentCommand.JustifyLeft;
     if (alignment == Alignment.Center) {
         command = DocumentCommand.JustifyCenter;
         align = 'center';
     } else if (alignment == Alignment.Right) {
         command = DocumentCommand.JustifyRight;
         align = 'right';
     }
-
-    editor.addUndoSnapshot(() => {
+    if (addUndoSnapshot) {
+        element.style.textAlign = align;
+    } else {
         execCommand(editor, command);
-        editor.queryElements(
-            '[align]',
-            QueryScope.OnSelection,
-            node => (node.style.textAlign = align)
-        );
-    }, ChangeSource.Format);
+    }
+}
+
+/**
+ * Check if the whole table is selected
+ * @param selection
+ * @returns
+ */
+function isWholeTableSelected(selection: TableSelectionRange) {

```

Comment:
what if I select more than whole table?

Reply:
Only the text is aligned. 
![tableAndText](https://user-images.githubusercontent.com/87443959/155421588-529570aa-efe3-4e17-94bd-3a3a44378281.gif)

Reply:
I think we should center-align both text outside the table and the table itself

Reply:
Ok, I checked the behavior or Word, it is the same with your current implementation. So I think it is fine here.